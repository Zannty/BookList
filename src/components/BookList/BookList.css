.current-page {
  font-weight: bold;
}
.pages {
  margin-left: 50em;
  margin-right: auto;
  letter-spacing: 0.3em;
  font-size: 20px;
}
.pages:hover {
  cursor: pointer;
}
