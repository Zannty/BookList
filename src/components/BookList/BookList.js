import React, { useEffect } from "react";
import "./BookList.css";
import BookListItem from "./../BookListItem/BookListItem";
import BookListService from "./../../services/BookListService";
import { connect } from "react-redux";
import { fetchRequest, fetchSuccess } from "./../../actions/action";
import Spinner from "./../Spinner/Spinner";

export const BookList = ({
  fetchRequest,
  fetchSuccess,
  loading,
  renderData
}) => {
  useEffect(() => {
    const bookListService = new BookListService();
    fetchRequest();
    bookListService.getBooks().then(data => fetchSuccess(data));
  }, []);

  if (loading) return <Spinner />;

  if (renderData.length === 0)
    return (
      <h2>
        К сожалению по вашему запросу ничего не найдено, попробуйте еще раз.
      </h2>
    );

  return (
    <ul className="list-group">
      {renderData.map(el => {
        return (
          <li key={el.id} className="list-group-item">
            <BookListItem book={el} />
          </li>
        );
      })}
    </ul>
  );
};
const mapStateToProps = ({ renderData, loading }) => {
  return {
    renderData,
    loading
  };
};

const mapDispatchToProps = dispatch => {
  return {
    fetchRequest: () => dispatch(fetchRequest()),
    fetchSuccess: data => dispatch(fetchSuccess(data))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(BookList);
