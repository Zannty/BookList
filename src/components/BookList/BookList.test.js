import React from "react";
import { BookList } from "./BookList";

describe("BookList", () => {
  const mockFn = jest.fn();
  let props = {
    fetchRequest: mockFn,
    loading: false,
    renderData: [{ id: 1, photo: "x", genre: "y", author: "z", name: "q" }]
  };
  let container;

  it("Spinner && loading=true", () => {
    props = { ...props, loading: true };
    container = shallow(<BookList {...props} />);
    expect(container.find("Spinner").debug()).toEqual("<Spinner />");
  });

  it("BookList snap", () => {
    props = { ...props, loading: false };
    container = mount(<BookList {...props} />);
    expect(container.debug()).toMatchSnapshot();
  });

  it("dispatch fetch called", () => {
    expect(mockFn).toHaveBeenCalled();
  });

  it("render data length === 0", () => {
    props = { ...props, renderData: [] };
    container = shallow(<BookList {...props} />);
    expect(container.props().children).toEqual(
      "К сожалению по вашему запросу ничего не найдено, попробуйте еще раз."
    );
  });
});
