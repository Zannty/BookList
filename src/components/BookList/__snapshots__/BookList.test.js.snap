// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList BookList snap 1`] = `
"<BookList fetchRequest={[Function: mockConstructor]} loading={false} renderData={{...}}>
  <ul className=\\"list-group\\">
    <li className=\\"list-group-item\\">
      <BookListItem book={{...}}>
        <div className=\\"book-list-item\\">
          <div className=\\"cover\\">
            <img src=\\"x\\" />
          </div>
          <div className=\\"property-item\\">
            <div>
              Название книги: 
              q
            </div>
            <div>
              Автор: 
              z
            </div>
            <div>
              Жанр: 
              y
            </div>
          </div>
        </div>
      </BookListItem>
    </li>
  </ul>
</BookList>"
`;
